### PR TITLE
adding exercise.coffee

### PR DESCRIPTION
**Pull request by achengscode**
Adding an exercise script to Cliobot, which lets a user ask Cliobot for an exercise and a quantity for that exercise

i.e. `@Cliobot exercise me` results in `@[username] give me 25 pushps`

### CODE REVIEW

File: scripts/exercise.coffee [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#   Prompts a user to do an exercise based on a random timer.
+#
+#   Commands:
+#     hubot exercise me - Join the exercise lottery!
+#
+#   Author:
+#     achengscode
+
+exercises = [ "pushups", "plank", "sit-ups", "wall sit", "lunges"]
+images = [ "https://media.giphy.com/media/87xihBthJ1DkA/giphy.gif",
```
Comment (michaelcheung) [inline scripts/exercise.coffee]: Can I trust you that all these images are SFW?

File: scripts/exercise.coffee [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+           "http://45.media.tumblr.com/bd0b35e9486e1092a836192ad8b3261b/tumblr_n37y2eaYLw1ry1rm7o8_400.gif",
+           "https://media.giphy.com/media/10BkwRHKkB1PPi/giphy.gif",
+           "https://media.giphy.com/media/53RTxXl37LWYU/giphy.gif",
+           "https://45.media.tumblr.com/1716e1f9f2a2fa68a8d0d61b0819b112/tumblr_nix40b3Jg71t55xupo1_500.gif",
+           "https://media.giphy.com/media/2OMHmoFMiJjfq/giphy.gif",
+           "https://media3.giphy.com/media/BvsKJXGzqfNPq/200.gif"
+]
+
+module.exports = (robot) ->
+  robot.respond /exercise me/i, (msg) ->
```
Comment (michaelcheung) [inline scripts/exercise.coffee]: Can we change the regex to `/exercise|punish me/i`?

File: scripts/exercise.coffee [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      when "wall sit" then 60
+
+    lower_limit = switch exercise
+      when "pushups", "sit-ups", "lunges" then 10
+      when "plank", "wall sit" then 60
+
+    repetitions = Math.floor Math.random() * (upper_limit - lower_limit) + lower_limit
+
+    response = switch exercise 
+      when "pushups", "sit-ups", "lunges" then "give me " + repetitions + " " + exercise
```
Comment (michaelcheung) [inline scripts/exercise.coffee]: `Give` here and below. You could also use string interpolation here if you wanted.

File: scripts/exercise.coffee (hunk 1 above)
Comment (achengscode) [inline scripts/exercise.coffee]: Yes. I can show you the images as well, but they are all SFW, in my opinion.

File: scripts/exercise.coffee (hunk 3 above)
Comment (achengscode) [inline scripts/exercise.coffee]: Ah okay. Forgot coffeescript had the ability to do that.

File: scripts/exercise.coffee (hunk 2 above)
Comment (achengscode) [inline scripts/exercise.coffee]: Yup, done!

File: scripts/exercise.coffee [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+           "https://45.media.tumblr.com/1716e1f9f2a2fa68a8d0d61b0819b112/tumblr_nix40b3Jg71t55xupo1_500.gif",
+           "https://media.giphy.com/media/2OMHmoFMiJjfq/giphy.gif",
+           "https://media3.giphy.com/media/BvsKJXGzqfNPq/200.gif",
+           "http://49.media.tumblr.com/9ba6b45df8631a3afc1128e26229b562/tumblr_nmpbp42gAF1up9ckeo1_500.gif",
+           "http://25.media.tumblr.com/aecf8a635799ab2657b5970b4bf79b4c/tumblr_mksyxu0IrV1r95jeco1_400.gif",
+           "http://24.media.tumblr.com/d35132701c13dfa7aa23e35ee5d51c52/tumblr_n5d0tlWgK81sbitb2o1_400.gif"
+]
+
+module.exports = (robot) ->
+  robot.respond /exercise|punish me/i, (msg) ->
```
Comment (michaelcheung) [inline scripts/exercise.coffee]: Sorry, it should be `/[exercise|punish] me/i` if you have to include the `me`.

Comment (achengscode) [inline scripts/exercise.coffee]: Shouldn't it be round brackets instead of square ones?

Comment (michaelcheung) [inline scripts/exercise.coffee]: You're right! I shouldn't look at regex late in the day.

**Pull request merged**